### PR TITLE
Feature/OpenAI responses api

### PR DESCRIPTION
**Pull request by LeXwDeX**
### Related GitHub Issue

Closes: # 

### Description

- Added an `openAiResponsesApiFormat` flag to provider settings so OpenAI-compatible providers can opt into the Responses API flow.
- Updated the OpenAI provider to route requests through the Responses API when the flag is enabled, covering both streaming and non-streaming paths and mapping usage metrics.
- Introduced `convertToResponsesApiMessages` along with lightweight typings for third-party diff/shell-quote helpers, ensuring type-safety across shared and webview code.
- Exposed a new toggle in the OpenAI-compatible settings UI with localized copy to help users enable Responses API mode for Azure-only models.

### Test Procedure

- `pnpm run lint`
- `pnpm run check-types`

### Pre-Submission Checklist

- [ ] **Issue Linked**
- [x] **Scope**
- [x] **Self-Review**
- [x] **Testing**
- [x] **Documentation Impact**
- [x] **Contribution Guidelines**

### Documentation Updates

- [x] No documentation updates are required.
- [ ] Yes, documentation updates are required. (Describe updates or link another PR.)

### Additional Notes

- The new Responses API toggle defaults to off to preserve existing behavior for users on the classic chat/completions endpoint.

---

image

image

----

> [!IMPORTANT]
> Adds OpenAI Responses API support with a new flag, updates provider logic, UI toggle, and type definitions.
> 
>   - **Behavior**:
>     - Adds `openAiResponsesApiFormat` flag in `provider-settings.ts` to enable OpenAI Responses API.
>     - Updates `OpenAiHandler` in `openai.ts` to use Responses API when the flag is enabled, handling both streaming and non-streaming requests.
>     - Introduces `convertToResponsesApiMessages` in `openai-format.ts` for message conversion.
>   - **UI**:
>     - Adds toggle for Responses API in `OpenAICompatible.tsx` with localized text in `settings.json` (English and Chinese).
>   - **Types**:
>     - Adds type definitions for `diff` and `shell-quote` modules in `diff.d.ts` and `shell-quote.d.ts`.
>   - **Misc**:
>     - Updates `parseUnifiedDiff.ts` to use `DiffHunk` type from `diff` module.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=RooCodeInc%2FRoo-Code&utm_source=github&utm_medium=referral)<sup> for 92120834d371b7ab519ba903230bf5c6bfaa2053. You can [customize](https://app.ellipsis.dev/RooCodeInc/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by ghost** [comment]
> **Rooviewer**  Clock &nbsp;&nbsp;[See task on Roo Cloud](https://app.roocode.com/cloud-jobs/BfIzlA3W?utm_source=github-comment&utm_medium=link&utm_campaign=github.pr.review)

Review completed. Found 3 issues that should be addressed:

- [ ] **Silent data loss in message conversion** - Images and tool_use blocks are dropped without warning in `convertToResponsesApiMessages`, potentially causing user confusion
- [ ] **System prompt handling** - System prompt is sent as user message instead of system-level instruction, which may affect model behavior
- [ ] **Missing test coverage** - New Responses API functionality lacks unit tests

<sub>Mention @roomote in a comment to request specific changes to this pull request or [fix all unresolved issues](https://app.roocode.com/fix-issues?repo=RooCodeInc/Roo-Code&prNumber=9255&commentId=3530621702).</sub>

File: src/api/transform/openai-format.ts [hunk 1]
```diff
@@ -1,6 +1,44 @@
...
+			}
+
+			if (textParts.length > 0) {
+				responseMessages.push({ role: anthropicMessage.role, content: textParts.join("\n") })
+			}
+		}
+	}
+
+	return responseMessages
+}
```
Comment (ghost) [inline src/api/transform/openai-format.ts:40]: The new `convertToResponsesApiMessages` function and the entire Responses API flow lack test coverage. This is a significant feature that handles message transformation and API interaction, and should have unit tests to verify correct behavior, especially around edge cases like empty messages, images, tool results, and the various content types. Consider adding tests similar to those that exist for `convertToOpenAiMessages`.

<sub>[Fix it with Roo Code](https://app.roocode.com/fix-issue/?repo=RooCodeInc/Roo-Code&prNumber=9255&fixId=c5e62d999e) or mention @roomote and request a fix.</sub>
